crates/tx-parser/src/filter.rs

```diff
@@ -1,92 +1,94 @@
-fn get_operator_wallet_pks(params: &RollupParams) -> Vec<Buf32> {
-    let OperatorConfig::Static(operator_table) = &params.operator_config;
-
-    operator_table.iter().map(|op| *op.wallet_pk()).collect()
-}
-
-pub fn derive_tx_filter_rules(params: &RollupParams) -> anyhow::Result<Vec<TxFilterRule>> {
-    let operator_wallet_pks = get_operator_wallet_pks(params);
-    let address = generate_taproot_address(&operator_wallet_pks, params.network)?;
-    let deposit_provider = params.get_deposit_params(address);
-    Ok(vec![
-        TxFilterRule::RollupInscription(params.rollup_name.clone()),
-        TxFilterRule::DepositRequest(deposit_provider.clone()),
-        TxFilterRule::Deposit(deposit_provider),
-    ])
-}
-/// Filter all the relevant [`Transaction`]s in a block based on given [`TxFilterRule`]s
-pub fn filter_relevant_txs(block: &Block, filters: &[TxFilterRule]) -> Vec<ProtocolOpTxRef> {
+/// Filter protocol operations as refs from relevant [`Transaction`]s in a block based on given
+/// [`TxFilterConfig`]s
+pub fn filter_protocol_op_tx_refs(
+    block: &Block,
+    filter_config: TxFilterConfig,
+) -> Vec<ProtocolOpTxRef> {
     block
         .txdata
         .iter()
         .enumerate()
         .filter_map(|(i, tx)| {
-            check_and_extract_relevant_info(tx, filters)
+            extract_protocol_op(tx, &filter_config)
                 .map(|relevant_tx| ProtocolOpTxRef::new(i as u32, relevant_tx))
         })
         .collect()
 }
 
 ///  if a [`Transaction`] is relevant based on given [`RelevantTxType`]s then we extract relevant
-///  info
-fn check_and_extract_relevant_info(
+///  info.
+///  TODO: make this function return multiple ops as a single tx can have multiple outpoints that's
+///  relevant
+fn extract_protocol_op(
     tx: &Transaction,
-    filters: &[TxFilterRule],
+    filter_conf: &TxFilterConfig,
 ) -> Option<ProtocolOperation> {
-    filters.iter().find_map(|rel_type| match rel_type {
-        TxFilterRule::RollupInscription(name) => {
-            if !tx.input.is_empty() {
-                if let Some(scr) = tx.input[0].witness.tapscript() {
-                    if let Ok(inscription_data) = parse_inscription_data(&scr.into(), name) {
-                        if let Ok(signed_batch) = borsh::from_slice::<SignedBatchCheckpoint>(
-                            inscription_data.batch_data(),
-                        ) {
-                            return Some(ProtocolOperation::RollupInscription(signed_batch));
-                        }
-                    }
-                }
-            }
-            None
-        }
+    // Currently all we have are inscription txs, deposits and deposit requests
+    parse_inscription_checkpoint(tx, filter_conf)
+        .map(ProtocolOperation::Checkpoint)
+        .or_else(|| parse_deposit(tx, filter_conf).map(ProtocolOperation::Deposit))
+        .or_else(|| parse_deposit_request(tx, filter_conf).map(ProtocolOperation::DepositRequest))
```

Comment:
I really would like to encapsulate the fact that we are parsing first DTs then DRTs in another place rather than the ordering of `or_else` in a `extract_foo` function.
Can't we move this logic elsewhere?

Reply:
Maybe below in a `parse` function that uses these sequentially in a proper manner/ordering?

Reply:
Do you mean, a single transaction can be both DT and DRT? If so then we have a [ticket](https://alpenlabs.atlassian.net/browse/STR-402) that says we should be able to parse multiple items from a transaction. So, the changes you suggested can possible done while working on that ticket.

Reply:
We should be counting *all* of the ops that match, not just the first.